engine/apps/alerts/models/alert_group.py

```diff
@@ -1363,38 +1630,64 @@ def _bulk_silence(user: User, alert_groups_to_silence: "QuerySet[AlertGroup]", s
-                resolved_by_user=None,
-                resolved_by=AlertGroup.NOT_YET,
-                silenced=True,
-                silenced_at=now,
-                silenced_until=silenced_until,
-                silenced_by_user=user,
-            )
-        else:
-            alert_groups_to_silence.update(
-                acknowledged=False,
-                acknowledged_at=None,
-                acknowledged_by_user=None,
-                acknowledged_by=AlertGroup.NOT_YET,
-                resolved=False,
-                resolved_at=None,
-                resolved_by_user=None,
-                resolved_by=AlertGroup.NOT_YET,
-                silenced=True,
-                silenced_at=now,
-                silenced_until=silenced_until,
-                silenced_by_user=user,
-                is_escalation_finished=True,
+        metrics_response_time = {}
+        metrics_states_diff = {}
+
+        for alert_group in alert_groups_to_silence_list:
+            new_response_time = alert_group.get_new_response_time()
+
+            # collect metrics
+            metrics_response_time, metrics_states_diff = MetricsCacheManager.metrics_update_diff_for_alert_group(
+                integration_id=alert_group.channel_id,
+                response_time_diff=metrics_response_time,
+                states_diff=metrics_states_diff,
+                old_state=alert_group.state,
+                new_state=STATE_SILENCED,
+                response_time=new_response_time,
+                started_at=alert_group.started_at,
             )
 
+            # update alert_group fields
+            if new_response_time:
+                alert_group.response_time = new_response_time
+            alert_group.acknowledged = False
+            alert_group.acknowledged_at = None
+            alert_group.acknowledged_by_user = None
+            alert_group.acknowledged_by = AlertGroup.NOT_YET
+            alert_group.resolved = False
+            alert_group.resolved_at = None
+            alert_group.resolved_by_user = None
+            alert_group.resolved_by = AlertGroup.NOT_YET
+            alert_group.silenced = True
+            alert_group.silenced_at = now
+            alert_group.silenced_until = silenced_until
+            alert_group.silenced_by_user = user
+            if not silence_for_period:
+                alert_group.is_escalation_finished = True
+
+        fields_to_update = [
+            "acknowledged",
+            "acknowledged_at",
+            "acknowledged_by_user",
+            "acknowledged_by",
+            "resolved",
+            "resolved_at",
+            "resolved_by_user",
+            "resolved_by",
+            "silenced",
+            "silenced_at",
+            "silenced_until",
+            "silenced_by_user",
+            "is_escalation_finished",
+            "response_time",
+        ]
+
+        AlertGroup.all_objects.bulk_update(alert_groups_to_silence_list, fields_to_update, batch_size=1000)
```

Comment:
there appears to be 5 other spots in this file's PR diff where you're modifying the object's fields like this. is it possible to deduplicate this into a method on the `AlertGroup` model class?

Reply:
I'm not sure how to do it better, since these methods update fields to different values, `fields_to_update` is also different in some places 🤔